File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 2/4_least_common_multiple/lcm.cpp
#include <iostream>
using namespace std;

long long EuclidGCD(long long a, long long b){
    if (b == 0){
        return a;
    }
    long long rem = a % b;
    return EuclidGCD(b, rem);
}

long long lcm(long long a, long long b){
    long long gcd = EuclidGCD(a, b);
    long long num = a / gcd;
    return num*b;
}

int main(){
    long long a, b;
    cin >> a >> b;
    cout << lcm(a, b);
    return 0;
}

File: Data-Structures-and-Algorithms-Specialization/1. Algorithmic Toolbox/Week 2/4_least_common_multiple/lcm.py
def EuclidGCD(a, b):
    if b == 0:
        return a
    
    rem = a % b
    return EuclidGCD(b, rem)

def lcm(a, b):
    gcd = EuclidGCD(a, b)
    num = a / gcd
    return int(num*b)


if __name__ == '__main__':
    a, b = map(int, input().split())
    print(lcm(a, b))

